core-bundle/src/Command/ResizeImagesCommand.php

```diff
@@ -0,0 +1,243 @@
+        $throttle = (float) $input->getOption('throttle');
+        $concurrent = (string) $input->getOption('concurrent');
+
+        if ($timeLimit < 0) {
+            throw new InvalidArgumentException(sprintf('Time-limit value "%s" is invalid.', $timeLimit));
+        }
+
+        if ($throttle < 0.001 || $throttle > 1) {
+            throw new InvalidArgumentException(sprintf('Throttle value "%s" is invalid.', $throttle));
+        }
+
+        if ((string) (int) $concurrent !== $concurrent || (int) $concurrent < 1) {
+            throw new InvalidArgumentException(sprintf('Concurrent value "%s" is invalid.', $concurrent));
+        }
+
+        if ((int) $concurrent > 1) {
+            return $this->executeConcurrent((int) $concurrent, $input, $output);
+        }
+
+        $startTime = microtime(true);
+
+        $maxDuration = 0;
+        $lastDuration = 0;
+
+        foreach ($this->storage->listPaths() as $path) {
+            $sleep = (1 - $throttle) / $throttle * $lastDuration;
+
+            if ($timeLimit && microtime(true) - $startTime + $maxDuration + $sleep > $timeLimit) {
+                $output->writeln("\n".'Time limit of '.$timeLimit.' seconds reached.');
+
+                return 0;
+            }
+
+            usleep((int) ($sleep * 1000000));
+
+            $maxDuration = max($maxDuration, $lastDuration = $this->resizeImage($path, $output));
+        }
+
+        $output->writeln("\n".'All images resized.');
+
+        return 0;
+    }
+
+    private function resizeImage(string $path, OutputInterface $output): float
+    {
+        $startTime = microtime(true);
+
+        if ($this->filesystem->exists($path)) {
+            return 0;
+        }
+
+        $output->write(str_pad($path, $this->terminalWidth - 14, '.').' ');
+
+        try {
+            $image = $this->imageFactory->create($this->targetDir.'/'.$path);
+            $resizer = $this->resizer;
+
+            if ($image instanceof DeferredImageInterface) {
+                $resizedImage = $resizer->resizeDeferredImage($image, false);
+
+                if (null === $resizedImage) {
+                    $output->write("\r".str_repeat(' ', $this->terminalWidth)."\r");
+                } else {
+                    $output->writeln(sprintf('done%7.3Fs', $duration = microtime(true) - $startTime));
+
+                    return $duration;
+                }
+            } else {
+                $output->write("\r".str_repeat(' ', $this->terminalWidth)."\r");
+            }
+        } catch (\Throwable $exception) {
+            $output->writeln('failed');
+            $output->writeln($exception->getMessage());
+        }
+
+        return 0;
+    }
+
+    private function executeConcurrent(int $count, InputInterface $input, OutputInterface $output): int
+    {
+        $phpFinder = new PhpExecutableFinder();
+
+        $output->writeln('Starting '.$count.' concurrent processes...');
+
+        if (false === ($phpPath = $phpFinder->find())) {
```

Comment:
I think this could be confusing. It’s IMO better to get the error message `The php executable could not be found.` instead of `The "--concurrent" option does not exist.`